src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -237,8 +237,12 @@ private function addFormSection(ArrayNodeDefinition $rootNode, callable $enableI
             ->children()
                 ->arrayNode('form')
                     ->info('Form configuration')
-                    ->{$enableIfStandalone('symfony/form', Form::class)}()
+                    ->treatFalseLike(['enabled' => false])
+                    ->treatTrueLike(['enabled' => true])
+                    ->treatNullLike(['enabled' => true])
+                    ->addDefaultsIfNotSet()
                     ->children()
+                        ->scalarNode('enabled')->defaultNull()->end() // defaults to !class_exists(FullStack::class) && class_exists(Form::class)
```

Comment:
why changing this one ? enabling form based on the presence of the form component was right here. This is what `$enableIfStandalone` is doing. and the `canBeEnabled` and `canBeDisabled` methods of symfony/config have an additional behavior: setting any _other_ setting implicitly enables it (losing that behavior is the reason why you had to update tests)

Reply:
because of https://github.com/symfony/recipes/pull/1337/files, which requires being able to set some defaults without enabling form while symfony/form isn't installed
alternatives are way too complex (recipes relying on many components being installed is not possible)

Reply:
I'm in favor of a specific csrf recipe as well. We do this for other optional framework config options as well (e.g. messenger, mailer)